### ISSUE TITLE
`poetry --directory […] run -- pwd` shows current directory

### ISSUE THREAD

**Issue by l0b0** [python-poetry/poetry#7363]
- **Poetry version**: Poetry (version 1.3.0)
- **Python version**: 3.9.16
- **OS version and name**: NixOS 22.11
- **pyproject.toml**: **N/A**

- [ ] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) stable Poetry version, installed using a recommended method.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] I have consulted the [FAQ](https://python-poetry.org/docs/faq/) and [blog](https://python-poetry.org/blog/) for any relevant entries or release notes.
- [ ] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option) and have included the output below. **N/A**

## Issue

Simplest possible reproduction:

```bash
cd "$(mktemp --directory)"
mkdir test
cd test
yes '' | poetry init
cd ..
poetry --directory test run pwd
```

Expected output: `[Temporary directory path]/test`

Actual output: `[Temporary directory path]`

In other words, it looks like the Poetry internals run within the subdirectory, but not the command actually called by `poetry run`.

**Comment by finswimmer** [comment]
Hello @l0b0,

this is expected. Poetry just loads the `pyproject.toml` from the given directory and does not switch to the directory.

If you think we can improve the doc about this, feel free to open a PR for it :smiley: 

fin swimmer

**Comment by l0b0** [comment]
The semantics of `-C`/`--directory` is different from other tools like GNU Make, and `poetry --help` says

> The working directory for the Poetry command

, so this is definitely not expected. I wouldn't want to open a PR to fix the doc, since I believe it shouldn't behave like this.
